uwblueprint/sistering
# Build create posting basic info form and page

## Ticket link

Closes #57



## Implementation description
* Implemented posting basic info form and page according to designs
* Wrote form data to global posting context
* Updated posting context to store full branch, skills, and employees objects
* Minor code cleanup

Create posting page:
<p float="left">
  <img src="https://user-images.githubusercontent.com/30205929/156906051-d34f1db2-d4c9-46e9-b5f0-a0a3d9fe6ea8.png" width="380" />
<img src="https://user-images.githubusercontent.com/30205929/156906132-2a15a3ba-268f-4f87-9311-83c3b758e93a.png" width="380" />
</p>
Review posting page (updated to display branch name and employee details):
<img src="https://user-images.githubusercontent.com/30205929/156906186-e0d7d679-9617-4c71-8892-dcf0d3a96989.png" width="380" />




## Steps to test
1. Click on the deploy preview link, sign in using `sistering+admin` credentials (staging env has been populated with seed data)
2. Click on `Create Posting` on the default page
3. Fill out form, omit some required fields
4. Click `Next`, verify that error messages are displayed
5. Complete the form correctly and click `Next` again
6. Click `Go Back`, then click `Review Posting`
7. Verify that the displayed details of the posting are correct



## What should reviewers focus on?
* `frontend/src/components/admin/posting/CreatePostingBasicInfo.tsx`
* Any part that can be simplified or refactored?


## Checklist
